File: docs/examples/loc_scale_m.py
import robstattm_py as rpm

# Zinc concentration in 53 mineral samples (some are gross outliers).
zinc = rpm.datasets.mineral()["zinc"].to_numpy()

# Robust M-estimators of location AND scale, in one call.
est = rpm.loc_scale_m(zinc, psi="bisquare", eff=0.95)

print(f"robust location: {est.mu:.4f}")
print(f"robust scale:    {est.disper:.4f}")
print(f"std-err of mu:   {est.std_mu:.4f}")
print()
print(f"compare classical: mean={zinc.mean():.4f}   std={zinc.std(ddof=1):.4f}")
print("(the classical mean/std are pulled by the outlying samples)")

File: docs/examples/loc_scale_m.R
data(mineral)
zinc <- mineral$zinc          # zinc concentration in 53 mineral samples

# Robust M-estimators of location AND scale, in one call.
est <- locScaleM(zinc, psi = "bisquare", eff = 0.95)

cat("robust location:", round(est$mu, 4), "\n")
cat("robust scale:   ", round(est$disper, 4), "\n")
cat("std-err of mu:  ", round(est$std.mu, 4), "\n")
